From 23c2ba084da0ab1514ae0487486ae6e5ba28963a Mon Sep 17 00:00:00 2001
From: Andy Yan <andy.yan@rock-chips.com>
Date: Wed, 13 Jun 2018 14:37:57 +0800
Subject: [PATCH] Use standard lz4 format for Image.lz4 on ARM64

The lz4 Legacy format(which specified by -l) is
not supported by U-Boot.

Change-Id: I6b94881117b59384daca4efd796c933e8dc9e5a6
Signed-off-by: Andy Yan <andy.yan@rock-chips.com>
---
 arch/arm64/boot/Makefile | 2 +-
 scripts/Makefile.lib     | 5 +++++
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/Makefile b/arch/arm64/boot/Makefile
index 7ab8e74cd83a..9eac75047996 100644
--- a/arch/arm64/boot/Makefile
+++ b/arch/arm64/boot/Makefile
@@ -39,7 +39,7 @@ $(obj)/Image.gz: $(obj)/Image FORCE
 	$(call if_changed,gzip)
 
 $(obj)/Image.lz4: $(obj)/Image FORCE
-	$(call if_changed,lz4)
+	$(call if_changed,lz4c)
 
 $(obj)/Image.lzma: $(obj)/Image FORCE
 	$(call if_changed,lzma)
diff --git a/scripts/Makefile.lib b/scripts/Makefile.lib
index b587abfb489a..1d0e885628e6 100644
--- a/scripts/Makefile.lib
+++ b/scripts/Makefile.lib
@@ -364,6 +364,11 @@ cmd_lz4 = (cat $(filter-out FORCE,$^) | \
 	lz4c -l -c1 stdin stdout && $(call size_append, $(filter-out FORCE,$^))) > $@ || \
 	(rm -f $@ ; false)
 
+quiet_cmd_lz4c = LZ4C	$@
+cmd_lz4c = (cat $(filter-out FORCE,$^) | \
+	lz4c -c1 stdin stdout && $(call size_append, $(filter-out FORCE,$^))) > $@ || \
+	(rm -f $@ ; false)
+
 # U-Boot mkimage
 # ---------------------------------------------------------------------------
 
-- 
2.35.3

